File: PRACTICA1/JAVA/deepspace/src/deepspace/Hangar.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;
import java.util.ArrayList;
/**
 *
 * @author Lucía Salamanca López
 *         Celia Arias Martínez
 */
public class Hangar {
    private int maxElements;
    private ArrayList<ShieldBooster> shieldBoosters;
    private ArrayList<Weapon> weapons;
    
    Hangar(int capacity){
        maxElements = capacity;
    }
    
    //CAMBIO hago new del array list de weapon
    Hangar(Hangar h){
       System.out.println(h.maxElements);
        maxElements = h.maxElements;
        shieldBoosters = new ArrayList<>();
        if (h.shieldBoosters != null){
            for(int i=0; i<h.shieldBoosters.size(); i++){
                shieldBoosters.add(h.shieldBoosters.get(i));
            }
        }
        weapons = new ArrayList<>();
        if (h.weapons!=null){
            for(int j=0; j<h.weapons.size(); j++){
                weapons.add(h.weapons.get(j));
            }
        }
    }
    HangarToUI getUIversion(){
        return new HangarToUI(this);
    }
    
    public boolean spaceAvailable(){
       if ((shieldBoosters.size()+weapons.size()) < maxElements)
               return true;
       return false;
    }
    
    
    public boolean addWeapon(Weapon w){
        if(spaceAvailable()){
            weapons.add(w);
            return true;
        }
        return false;
    }
    
    public boolean addShieldBooster(ShieldBooster s){
         if(spaceAvailable()){
            shieldBoosters.add(s);
            return true;
        }
        return false;
    }
    public int getMaxElements(){
        return maxElements;
    }
    public ArrayList<ShieldBooster> getShieldBoosters(){
        return shieldBoosters;
    }
    //cambio Weapon[] por Arraylist
    public ArrayList<Weapon> getWeapons(){
        return weapons;
    }
    public ShieldBooster removeShieldBooster(int s){
          if ((s<0) || (s > (shieldBoosters.size()-1)))
            return null;
        return shieldBoosters.remove(s);
    }
    public Weapon removeWeapon(int w){
        if ((w<0) || (w > (weapons.size()-1)))
            return null;
        return weapons.remove(w);
    }
    
    
       
}

File: PRACTICA1/RUBY/Deepspace/Hangar.rb
#encoding:utf-8
require './HangarToUI'
module Deepspace


# @author Lucía Salamanca López, Celia Arias Martínez
class Hangar

def initialize(capacity)
  @maxElements = capacity
  @weapons = Array.new
  @shieldBoosters = Array.new

end

def self.newCopy(h)
  nuevo = new(h.maxElements)
  if (@weapons != nil)
    for w in @weapons
      nuevo.addWeapon(w)
    end 
  end 
  if (@shieldBoosters!=nil)
    for sb in @shieldBoosters
      nuevo.addShieldBooster(sb)
    end 
  end 
  return nuevo
end

def getUIversion
  HangarToUI.new(self)
end

def spaceAvailable()
  (@weapons.length + @shieldBoosters.length) < @maxElements
end

def addWeapon(w)
  if (spaceAvailable)
    @weapons << w
    return true
  else
    return false
  end
end

def addShieldBooster (s)
  if (spaceAvailable)
    @shieldBoosters << s
    return true
  else
    return false
  end
end

def maxElements
  @maxElements
end

def shieldBoosters
  @shieldBoosters
end

def weapons
  @weapons
end

def removeShieldBooster(s)
  if (@shieldBoosters[s] != nil)
    potenEscudo = @shieldBoosters[s]
    @shieldBoosters.delete_at(s)
    return potenEscudo
  else
    return nil
  end
end

def removeWeapon(w)
  if (@weapons[w] != nil)
    arma = @weapons[w]
    @weapons.delete_at(w)
    return arma
  else
    return nil
  end
end


def to_s
  return "Max Elements: #{@maxElements}, Weapons: [#{@weapons.join(", ")}], Shields: [#{@shieldBoosters.join(", ")}]"
end

end
end
